getpaid/backends/pagseguro/__init__.py

```diff
@@ -0,0 +1,204 @@
+        
+        self._PRODUCT_DATA.update(itemId1=self.payment.id,
+                                  itemDescription1= product_description.encode("latin1", "ignore"), 
+                                  currency=self.payment.currency,
+                                  itemAmount1=self.payment.amount,
+                                  reference=reference,
+                                  redirectURL=redirectURL,
+                                  )
+
+        full_data = dict(self._ACOUNT_DATA, **self._PRODUCT_DATA)
+        params = urlencode(full_data)
+
+        payment_full_url = "%s%s" % (gateway_url, params)
+        request.encoding = 'ISO-8859-1'
+        
+        logger.info(payment_full_url)
+
+        headers = {"Content-type": "application/x-www-form-urlencoded"}
+        response = requests.post(payment_full_url, headers=headers).text
+
+        code = ""
+        logger.info("xml response from pagseguro: " + str(response.encode("utf-8")))
+        dom = parseString(response)
+        
+        checkout = dom.getElementsByTagName("checkout")
+        if checkout:
+            childNodes = checkout[0].childNodes
+            code = childNodes[0].firstChild.nodeValue
+            date = childNodes[1].firstChild.nodeValue
+
+        self.payment.external_id = reference
+        self.payment.description = code
+        self.payment.save();
+        
+        return "%s%s" % (gateway_url2, code), 'GET', {}
+
+    @staticmethod
+    def process_notification(params):
+
+        payp = PaymentProcessor
+        tmp_transaction_url = payp.get_backend_setting('transaction_url',
+                                            payp._TRANSACTION_CONSULT_URL)
+        
+        is_dev = payp.get_backend_setting('testing', False)
+        if is_dev:
+            tmp_transaction_url = payp.get_backend_setting('dev_transaction_url',
+                                            payp._DEV_TRANSACTION_CONSULT_URL)
+
+        transaction_url = tmp_transaction_url + "%s?email=%s&token=%s"
+
+        Payment = get_model('getpaid', 'Payment')
+        
+        token = payp.get_backend_setting('token')
+        email = payp.get_backend_setting('email')
+
+        url_consult = transaction_url % (params["notificationCode"], email, token)
+        resp = requests.get(url_consult)
+        logger.info("pagseguro notification: " + resp.text)
+
+        dom = parseString(resp.text.encode('utf-8'))
+        transactionNode = dom.getElementsByTagName("transaction")
+        code = dom.getElementsByTagName("code")[0].firstChild.nodeValue
+        if dom.getElementsByTagName("status"):
+            status_code = int(dom.getElementsByTagName("status")[0].firstChild.nodeValue)
+        
+        reference = ""
+        reference_node = dom.getElementsByTagName("reference")
+        if reference_node:
+            reference = reference_node[0].firstChild.nodeValue
+        
+        try:
+            payment = Payment.objects.get(external_id__exact=reference)
```

Comment:
Payment.external_id can be blank and None so you really need to add an `if reference` here.